operator/pkg/controllers/master/kubeapiserver.go

```diff
@@ -84,7 +84,8 @@ func apiServerPodSpecFor(controlPlane *v1alpha1.ControlPlane) v1.PodSpec {
 		HostNetwork:                   true,
 		DNSPolicy:                     v1.DNSClusterFirstWithHostNet,
 		PriorityClassName:             "system-cluster-critical",
-		NodeSelector:                  nodeSelector(controlPlane.ClusterName()),
+		NodeSelector:                  nodeSelector(controlPlane.ClusterName(), controlPlane.Spec.ColocateAPIServerWithEtcd),
+		Affinity:                      affinity(controlPlane.Spec.ColocateAPIServerWithEtcd),
```

Comment:
Remind me why do we need Affinity for apiserver component and not for other components like KCM/Scheduler/EncryptionProvider/Authenticator ?

Reply:
They are running as daemonsets